### ISSUE TITLE
3D construction and texturing is not right.

### ISSUE THREAD

**Issue by sbasnagala** [evo-biomech/scAnt#12]
I have successfully run the Meshroom, but resulted obj file seems not right. It is supposed to be a beetle, but it lacks the basic structure. I had 169 resulting stacked images in Meshroom to run the model. 10 images out of 169 had no solid background. Any idea what's happening
![Beetle-2022-08-03 150749](https://user-images.githubusercontent.com/102258948/182689855-4b615ed1-d71a-4729-9108-c5f173425cbf.jpg)
?

![image](https://user-images.githubusercontent.com/102258948/182690100-ca289ec3-6d1a-438f-a8d0-273a64e92450.png)

![image](https://user-images.githubusercontent.com/102258948/182690307-421ea58a-bca9-4337-bab9-ba6d0ae079fe.png)

**Comment by mnhn-paul** [comment]
@sbasnagala The cameras seem to be aligned correctly. did the dense point cloud calculate correctly?
I think this might be an issue with your Meshroom workflow.

you might have more luck posting you issue to an Meshroom or Photogrammetry groups on Reddit or so.

**Comment by sbasnagala** [comment]
I did post in their forum.
Thanks

**Comment by sbasnagala** [comment]
Hi Fabian!

During one of the discussion, you mentioned....

**"Keep in mind that you may need to adjust further parameters to fit your images, such as the min/max threshold values, as well as CLAHE settings, the same way you would in the scAnt GUI."**

Could you tell me where exactly I can change these settings? 
Thanks
sbasnagala

**Comment by FabianPlum** [comment]
Hi Suranga,

There are **three** ways in which you can adjust these values:

- in the **GUI** (BEFORE scanning) : the values can be changed in the scanning settings 
- in the **config file** (BEFORE scanning): make changes to the `mask_thresh_min` / `mask_thresh_max` values in the config file and load it as a preset before starting a scan
- as a post-processing step (when running masking): when running the command line version of the masking script you need to define the mask_thresh_min / mask_thresh_max as well as **CLAHE** values directly. The darkest background pixel value across your RGB channels is the  `mask_thresh_min` value, and the brightest is the `mask_thresh_max`value. Your CLAHE value should be chosen based on how clearly separated your specimen is from the image background. Use low values (1 to 3) for well-separated fine features, and high values (3 to 10) for less sharp coarse features. 

e.g:

`python mask_generator.py --images "PATH_TO_YOUR_STACKED_IAMGES" --CLAHE 3 --mask_thresh_min 175 --mask_thresh_max 215`

All the best
Fabi

**Comment by FabianPlum** [comment]
As the questions are no longer related to the use of Meshroom and I cannot give specific advice with third-party software I will need to close this issue here. Feel free to open a new issue with further questions relating to the scanning process itself!
